nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -211,17 +212,52 @@ def make_dataset(self, data_dimension: tp.Optional[int], dataset: str) -> None:
                     raise ImportError("Please install keras (pip install keras) to use keras ml tuning") from e
 
                 data = keras.datasets.boston_housing
+            elif dataset == "auto-mpg":
+                mpg_url="http://www-lisic.univ-littoral.fr/~teytaud/files/Cours/Apprentissage/data/auto-mpg.data"
+                data = pd.read_csv(mpg_url)
+            elif dataset == "red-wine":
+                red_url="http://www-lisic.univ-littoral.fr/~teytaud/files/Cours/Apprentissage/data/winequality-red.csv"
+                data = pd.read_csv(red_url)
+            elif dataset == "white-wine":
+                white_url="http://www-lisic.univ-littoral.fr/~teytaud/files/Cours/Apprentissage/data/winequality-white.csv"
+                data = pd.read_csv(white_url)
             else:
                 data = {"boston": sklearn.datasets.load_boston,
                         "diabetes": sklearn.datasets.load_diabetes,
                         }[dataset](return_X_y=True)
 
             # Half the dataset for training.
+            test_ratio = 0.5
             if dataset == "kerasBoston":
-                (self.X_train, self.y_train), (self.X_test, self.y_test) = data.load_data(test_split=0.5, seed=42)
+                (self.X_train, self.y_train), (self.X_test, self.y_test) = data.load_data(test_split=test_ratio, seed=42)
+            elif dataset == "auto-mpg":
+                data.drop('name', 1, inplace=True)
+                train, test = train_test_split(data, test_size=test_ratio)
+                self.X_train = train
+                self.y_train = train['mpg']
+                del(self.X_train['mpg'])
+                self.X_test = test
+                self.y_test = test['mpg']
+                del(self.X_test['mpg'])
+                self.X_train = self.X_train.to_numpy()
+                self.X_test = self.X_test.to_numpy()
+                self.y_train = self.y_train.to_numpy()
+                self.y_test = self.y_test.to_numpy()
+            elif dataset == "red-wine" or dataset == "white-wine":
+                train, test = train_test_split(data, test_size=test_ratio)
+                self.X_train = train
+                self.y_train = train['quality']
+                del(self.X_train['quality'])
+                self.X_test = test
+                self.y_test = test['quality']
+                del(self.X_test['quality'])
+                self.X_train = self.X_train.to_numpy()
+                self.X_test = self.X_test.to_numpy()
+                self.y_train = self.y_train.to_numpy()
+                self.y_test = self.y_test.to_numpy()
```

Comment:
same thing here, please find a way to factorize this, the less duplicated code the less maintenance we'll need (and there is always way too much maintenance to do  :s )

A few remarks in the meantime:
- `del` is a keyword not a function so I think it should be: `del self.X_train['quality']`
- you are assigning dataframes to data numpy arrays, so typing is erroneous (although `mypy` will not complain because typing of pandas is lacking so it's considered as "Any"/unknown)

Reply:
You're right, I hope it is better now.

Reply:
perfect ;) thanks